source/slang/slang-ir.cpp

```diff
@@ -2821,6 +2837,16 @@ namespace Slang
         return structType;
     }
 
+    IRAssociatedType* IRBuilder::createAssociatedType()
```

Comment:
Given that `IRAssociatedType` has no operands, it seems like we face the same decision point as with `IRThisType`:

1. We could have a single global instance of `IRAssociatedType`, and all references to associated types in the signature of an interface requirement will refer to that one instance.

2. We could have a distinct instance of `IRAssociatedType` for each interface and each associated type requirement on that interface. In that case, we could imagine `IRAssociatedType` having operands that name the interface and the requirement key.

For option (2), it might actually work really well if an interface type was a parent isntruction, and then an `IRAssociatedType` was a child of that interface, so that we could be careful that references to the associated type in this fashion are properly restricted to only occur in the scope of the interface's requirement signatures.

Going even deeper, it is possible that an `IRAssociatedType` could actually just *be* the `IRInterfaceRequirementEntry` that represents the associated type!

Reply:
Done.